File: EX1/Python/src/Matrix_Multiply.py

size=1000

matrix1 = [[i+1 for i in range(size)] for j in range(size)]
matrix2 = [[i+1 for i in range(size)] for j in range(size)]

ans = [[0 for i in range(size)] for j in range(size)]

for i in range(size):
    for j in range(size):
        tmp=0
        for n in range(size):
            tmp+=matrix1[i][n]*matrix2[n][j]
        ans[i][j]=tmp

print(ans)

File: EX1/Haskell/src/Matrix_Multiply.hs
import Data.List

multiplyMatrix :: Num x => [[x]] -> [[x]] -> [[x]]
multiplyMatrix x y = zipWith (zipWith (+)) x y


getMatrix n =
  let rowIndex n k = [x | x <- [1,2..n]]
  in [rowIndex n k | k <- [1,2..n]]

main = do
  let matrix_size = 1500::Int
  let matrix1 = getMatrix matrix_size --得到两个1500*1500的矩阵
  let matrix2 = getMatrix matrix_size

  let product = multiplyMatrix matrix1 matrix2
  print product
